web/src/components/task/task-sidebar/finish-button/finish-button.tsx

```diff
@@ -76,45 +89,70 @@ export const FinishButton: FC<TaskSidebarProps> = ({
-                                  handleFinishValidation();
-                              }
-                            : onAnnotationTaskFinish
-                    }
-                />
-            </Tooltip>
-            {isNextTaskPresented && (
-                <PickerInput
-                    valueType="id"
-                    minBodyWidth={322}
-                    selectionMode="single"
-                    value={redirectionSettings}
-                    dataSource={redirectSettingsDataSource}
-                    getName={(item) => item?.caption ?? ''}
-                    onValueChange={handleSetRedirectionSettings}
-                    renderToggler={({ toggleDropdownOpening, ...props }) => (
-                        <Button
-                            {...props}
-                            size="36"
-                            fill="solid"
-                            isDropdown={false}
-                            icon={settingsIcon}
-                            onClear={undefined}
-                            placeholder={undefined}
-                            isDisabled={isDisabled}
-                            onClick={toggleDropdownOpening}
-                        />
-                    )}
-                />
-            )}
-        </ControlGroup>
+        <>
+            <ControlGroup cx={styles['button-finish-control-group']}>
+                <Tooltip content={tooltipContent}>
+                    <Button
+                        isDisabled={isDisabled}
+                        cx={styles['button-finish']}
+                        caption={isValidation ? 'FINISH VALIDATION' : 'FINISH LABELING'}
+                        onClick={() => {
+                            if (confirmWindowNeed === 'true') {
+                                uuiModals
+                                    .show<string>((props) => <ConfirmModal {...props} />)
+                                    .then(() => {
+                                        isValidation
+                                            ? () => {
+                                                  handleFinishValidation();
+                                              }
+                                            : onAnnotationTaskFinish;
+                                    })
+                                    .catch(() => {
+                                        uuiNotifications
+                                            .show((props) => (
+                                                <WarningNotification {...props}>
+                                                    <FlexRow alignItems="center">
+                                                        <Text>Finishing aborted.</Text>
+                                                    </FlexRow>
+                                                </WarningNotification>
+                                            ))
+                                            .catch(() => null);
```

Comment:
In which case we may have some error here? In case if can't then let's remove it

Reply:
removed